### PR TITLE
Introduce new framework generation process

### PR DESCRIPTION
**Pull request by giginet**
This PR changes framework generation process.

### Before

Scipio tries to consist of frameworks with PIF. Modify PIF to generate framework  by injecting header and linker build phases.
But it breaks some packages. It's better to keep PIF as modified as possible.

### After

Just build framework components with xcbuild. After that, assemble framework bundles manually.

- PIF + xcbuild compiles binaries, Info.plist and swiftmodules in framework (Generated Framework)
- `FrameworkComponentsCollector` collects framework components from the generated framework.
- `FrameworkBundleAssembler` try to relocate them into framework bundle format (Assembled Framework)
- `XCBuildClient` tries to generate XCFramework from the assembled frameworks

### CODE REVIEW

File: Sources/ScipioKit/Producer/PIF/FrameworkComponentsCollector.swift [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        // Sometimes, public headers include a file and its symlink both.
+        // This situation raises a duplication error
+        // So duplicated symlinks have to be omitted
+        let notDuplicatedSymlinks = symlinks.filter { path in
+            notSymlinks.allSatisfy { FileManager.default.contentsEqual(atPath: path.pathString, andPath: $0.pathString) }
+        }
+            .map { $0.asURL.resolvingSymlinksInPath() }
+            .map(\.absolutePath)
+
+        return Set(notSymlinks + notDuplicatedSymlinks)
```
Comment (giginet) [inline Sources/ScipioKit/Producer/PIF/FrameworkComponentsCollector.swift:169]: This is the same logic `PIFGenerater` used to do this.

File: Tests/ScipioKitTests/InfoPlistGeneratorTests.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import Foundation
+import XCTest
+@testable import ScipioKit
+import TSCBasic
+
+final class InfoPlistGeneratorTests: XCTestCase {
```
Comment (giginet) [inline Tests/ScipioKitTests/InfoPlistGeneratorTests.swift:6]: This is a test case of an existing logic.

File: Sources/ScipioKit/Producer/PIF/FrameworkBundleAssembler.swift [hunk 3]
```diff
@@ -0,0 +1,121 @@
+import Foundation
+import TSCBasic
+
+/// A assembler to generate framework bundle
```
Comment (freddi-kit) [inline Sources/ScipioKit/Producer/PIF/FrameworkBundleAssembler.swift:4]:
```suggestion
```

It is already described in name of struct

File: Sources/ScipioKit/Producer/PIF/FrameworkComponentsCollector.swift [hunk 4]
```diff
@@ -0,0 +1,176 @@
+import Foundation
+import TSCBasic
+import PackageModel
+
+/// FileLists to assemble a framework bundle
+struct FrameworkComponents {
+    var name: String
```
Comment (freddi-kit) [inline Sources/ScipioKit/Producer/PIF/FrameworkComponentsCollector.swift:7]:
```suggestion
    var frameworkName: String
```

I'm not sure whether the name of "Component" or "Framework"

File: Sources/ScipioKit/Producer/PIF/FrameworkComponentsCollector.swift [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    }
+
+    func collectComponents(sdk: SDK) throws -> FrameworkComponents {
+        let modulemapGenerator = ModuleMapGenerator(
+            descriptionPackage: descriptionPackage,
+            fileSystem: fileSystem
+        )
+
+        // xcbuild automatically generates modulemaps. However, these are not for frameworks.
+        // Therefore, it's difficult to contain to final XCFrameworks.
```
Comment (freddi-kit) [inline Sources/ScipioKit/Producer/PIF/FrameworkComponentsCollector.swift:57]:
```suggestion
        // Therefore, it's difficult to contain this generated modulemaps to final XCFrameworks.
```

File: Sources/ScipioKit/Producer/PIF/FrameworkComponentsCollector.swift [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+            frameworkBundlePath.appending(components: "Resources", "Info.plist"),
+        ]
+        guard let infoPlistPath = infoPlistLocationCandidates.first(where: fileSystem.exists(_:)) else {
+            throw Error.infoPlistNotFound(frameworkBundlePath: frameworkBundlePath)
+        }
+        return infoPlistPath
+    }
+
+    /// Collects *.swiftmodules* in a generated framework bundle
+    private func collectSwiftModules(of targetName: String, in frameworkPath: AbsolutePath) throws -> AbsolutePath? {
```
Comment (freddi-kit) [inline Sources/ScipioKit/Producer/PIF/FrameworkComponentsCollector.swift:124]: What is difference of collect and find?

**Review by freddi-kit — APPROVED** [review]

File: Sources/ScipioKit/Producer/PIF/FrameworkBundleAssembler.swift (hunk 3 above)
Comment (giginet) [reply Sources/ScipioKit/Producer/PIF/FrameworkBundleAssembler.swift:4]: There is an additional doccomment following line. So it's better to remain this line.

File: Sources/ScipioKit/Producer/PIF/FrameworkComponentsCollector.swift (hunk 6 above)
Comment (giginet) [reply Sources/ScipioKit/Producer/PIF/FrameworkComponentsCollector.swift:124]: No, I'll unify them to collect.

**Pull request merged**